Line Display Options

MatthewBowles opened the pull request:
Added various options to change the appearance of lines in cut plots.
- Refactored legends code to be more extendable and removed redundant data
- Added controls to UI for line appearance ( line colour, style, width, marker type and show/hide)
- Add unit test to presenter for new line controls

**To test:**
- Open any cut plot
- Click plot options
- Change values for line colour / style / width / marker and click OK

Fixes #104 
Fixes #109

martyngigg commented on mslice/plotting/plot_window/plot_figure.py:
```diff
@@ -179,11 +161,75 @@ def _toggle_errorbars(self):
+                labels.append(legends[i]['label'])
+            self.legends_visible[i] = legends[i]['visible']
+        x = current_axes.legend(handles_to_show, labels)  # add new legends
+        x.draggable()
```
Good. It's useful being able to move the legend around.

MatthewBowles commented:
A scroll bar is now added to the legend options when there are three or more cuts. This was the simplest solution to implement.
Also, if a line colour is selected that is the same as an existing one, a warning message is displayed and it reverts to the previous selection.

MatthewBowles commented:
@martyngigg  I moved the legends to the right hand side as discussed, added the toggling for line visibility and made the show/hide controls a bit clearer.
Let me know if there is anything else that could be improved.

martyngigg commented:
@MatthewBowles Thanks.

I think this looks better but there seems to be a bit of a problem with the resize behaviour of the dialog. Does the UI file contain a layout?

![mslice-plot-options-resize](https://user-images.githubusercontent.com/733773/29356674-e3d14276-826c-11e7-82e3-6a6ce90e67b4.png)

The show behaviour looks nice and I've only found a minor issue:

* if I uncheck one of lines to show
* click okay
* click plot options again
* click "show legend" will just check the legend box but not cause "show" to become set - we should either keep the legend box disabled as before or enable show when show legend is clicked
